src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -2480,3 +2474,56 @@ std::string dbDotDecode(std::string s)
   std::replace(s.begin(), s.end(), ESCAPE_1_ENCODED, ESCAPE_1_DECODED);
   return s;
 }
+
+
+
+/* ****************************************************************************
+*
+* subscriptionsTreat -
+*
+* Lookup all subscriptions in the database and call a treat function for each
+*/
+void subscriptionsTreat(std::string tenant, OtisTreatFunction treatFunction)
+{
+  BSONObj                   query;
+  DBClientBase*             connection = getMongoConnection();
+  auto_ptr<DBClientCursor>  cursor;
+
+  try
+  {
+    cursor = connection->query(getSubscribeContextCollectionName(tenant).c_str(), query);
```

Comment:
Using this, you are quering for `{}` which means "all the subscriptions". I would be better to adjust this query to:

1. Only subscriptions with ONCHANGE
2. Only subscriptions which have at least one element using isPattern=true in the vector.

I will create a tech debt issue about refactor this function in order to use a narrower query.